src/sentry/uptime/subscriptions/subscriptions.py

```diff
@@ -306,25 +338,50 @@ def disable_project_uptime_subscription(uptime_monitor: ProjectUptimeSubscriptio
     # All project subscriptions are disabled, we can disable the subscription
     # and remove the remote subscription.
     if not has_active_subscription:
-        uptime_subscription.update(status=UptimeSubscription.Status.DISABLED.value)
+        uptime_subscription.status = UptimeSubscription.Status.DISABLED.value
+        uptime_subscription.save()
         delete_remote_uptime_subscription.delay(uptime_subscription.id)
 
 
-def enable_project_uptime_subscription(uptime_monitor: ProjectUptimeSubscription):
+def enable_project_uptime_subscription(
+    uptime_monitor: ProjectUptimeSubscription, ensure_assignment: bool = False
+):
     """
-    Re-enables a project uptime subscription. If the uptime subscription was
+    Enable a project uptime subscription. If the uptime subscription was
     also disabled it will be re-activated and the remote subscription will be
     published.
+
+    This method will attempt seat assignment via the quotas system. If There
+    are no available seats the monitor will be disabled and a
+    `UptimeMonitorNoSeatAvailable` will be raised.
+
+    By default if the monitor is already marked as ACTIVE this function is a
+    no-op. Pass `ensure_assignment=True` to force seat assignment.
     """
-    if uptime_monitor.status != ObjectStatus.DISABLED:
+    if not ensure_assignment and uptime_monitor.status != ObjectStatus.DISABLED:
         return
 
-    uptime_monitor.update(status=ObjectStatus.ACTIVE)
+    seat_assignment = quotas.backend.check_assign_seat(DataCategory.UPTIME, uptime_monitor)
+    if not seat_assignment.assignable:
+        disable_project_uptime_subscription(uptime_monitor)
+        raise UptimeMonitorNoSeatAvailable(seat_assignment)
+
+    outcome = quotas.backend.assign_seat(DataCategory.UPTIME, uptime_monitor)
+    if outcome != Outcome.ACCEPTED:
+        # Race condition, we were unable to assign the seat even though the
+        # earlier assignment check indicated assignability
+        disable_project_uptime_subscription(uptime_monitor)
+        raise UptimeMonitorNoSeatAvailable(None)
+
+    uptime_monitor.status = ObjectStatus.ACTIVE
+    uptime_monitor.save()
+
     uptime_subscription = uptime_monitor.uptime_subscription
 
     # The subscription was disabled, it can be re-activated now
     if uptime_subscription.status == UptimeSubscription.Status.DISABLED.value:
-        uptime_subscription.update(status=UptimeSubscription.Status.CREATING.value)
+        uptime_subscription.status = UptimeSubscription.Status.CREATING.value
+        uptime_subscription.save()
```

Comment:
What if the subscription isn't disabled but we need to re-link this to the project?

(Wondering if we should remove the unique constraint, and just have one uptime subscription per project. Less work than consolidating the tables, but might still be a lot to get done quickly...)

Reply:
ahh hmm, you might be right

Reply:
Actually it should be fine. We don't unlink everything. We just need to re-active the project uptime subscription and that will be good enough